internal/payloads/pubkey_payload.go

```diff
@@ -8,32 +8,54 @@ import (
 	"github.com/go-chi/render"
 )
 
-type PubkeyPayload struct {
-	*models.Pubkey
+// See models.Pubkey
+type PubkeyRequest struct {
+	Name string `json:"name" yaml:"name"`
+	Body string `json:"body" yaml:"body"`
+}
+
+// See models.Pubkey
+type PubkeyResponse struct {
+	ID                int64  `json:"id" yaml:"id"`
+	AccountID         int64  `json:"-" yaml:"-"`
+	Name              string `json:"name" yaml:"name"`
+	Body              string `json:"body" yaml:"body"`
+	Type              string `json:"type,omitempty" yaml:"type,omitempty"`
+	Fingerprint       string `json:"fingerprint,omitempty" yaml:"fingerprint,omitempty"`
+	FingerprintLegacy string `json:"fingerprint_legacy,omitempty" yaml:"fingerprint_legacy,omitempty"`
 }
-type (
-	PubkeyRequest  PubkeyPayload
-	PubkeyResponse PubkeyPayload
-)
 
 func (p *PubkeyRequest) Bind(_ *http.Request) error {
-	// Fingerprint is read-only field
-	p.Fingerprint = ""
 	return nil
 }
 
 func (p *PubkeyResponse) Render(_ http.ResponseWriter, _ *http.Request) error {
 	return nil
 }
 
+func (p *PubkeyRequest) NewModel() *models.Pubkey {
+	return &models.Pubkey{
+		Name: p.Name,
+		Body: p.Body,
+	}
+}
```

Comment:
Just curious, why did you choose to create a new model?

Reply:
This does not create new model, it returns a pointer of the current Pubkey model with two fields set, others are unset (thus default - empty string, 0 or nil). This is rather "conversion" function than anything else.